packages/components/src/Stack/Stack.tsx

```diff
@@ -21,17 +23,25 @@ export const Stack: React.FC<StackProps> = ({
   align = 'left',
   children,
   ...props
-}) => {
-  return (
-    <Box
-      {...props}
-      as={as}
-      display="flex"
-      flexDirection="column"
-      alignItems={ALIGNMENT[align]}
-      css={{ '> * + *': { pt: space } }}
-    >
-      {children}
-    </Box>
-  );
-};
+}) => (
+  <Box
+    {...props}
+    as={as}
+    display="flex"
+    flexDirection="column"
+    alignItems={ALIGNMENT[align]}
+    css={{ '> * + *': { pt: space } }}
+  >
+    {Children.map(
+      flattenChildren(children) as unknown as React.ReactElement,
+      (child: React.ReactElement) => {
+        if (as === 'div') {
```

Comment:
That you dont have an extra div coming with Box in line 40 inside the <ol> or <ul>

Reply:
Why do we need the extra div otherwise? Can you point me to the tests?

Reply:
Isn't it either a div or an list?

Reply:
Without the div we have the same problem like before. That the following class would be overwritten by base styles:
```
css-iw1nuy >*+* {
    padding-top: 16px;
}
```

Reply:
But y? I thought we wrap the children into `<div>` or `<li>` anyway 🤔

Reply:
At the moment you can give as children what you want. E.g.
```
<Stack space="medium" as="ol">
  <li>Milk</li>
  <li>Apples</li>
  <li>Cookies</li>
</Stack>
```
if you have Stack as div you can give for example a span as child. Actually we wrap this span with a div which has this padding-top style from above.

Reply:
Ok, but than you have to know as a marigold user that you dont use Stack as ol like this:
```
<Stack space="medium" as="ol">
  <li>Milk</li>
  <li>Apples</li>
  <li>Cookies</li>
</Stack>
```
Otherwise you have 
```
<Stack space="medium" as="ol">
  <li><li>Milk</li></li>
</Stack>
```
You should make as a user:
```
<Stack space="medium" as="ol">
  <span>Milk</span>
</Stack>
```

right?